homeassistant/components/unifiprotect/discovery.py

```diff
@@ -0,0 +1,66 @@
+"""The unifiprotect integration discovery."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+import logging
+from typing import Any
+
+from unifi_discovery import AIOUnifiScanner, UnifiDevice, UnifiService
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DISCOVERY = "discovery"
+DISCOVERY_INTERVAL = timedelta(minutes=60)
+
+
+async def async_start_discovery(hass: HomeAssistant) -> None:
+    """Start discovery."""
+    domain_data = hass.data.setdefault(DOMAIN, {})
+    if DISCOVERY in domain_data:
+        return
+    domain_data[DISCOVERY] = True
+
+    async def _async_discovery(*_: Any) -> None:
+        async_trigger_discovery(hass, await async_discover_devices(hass))
+
+    # Do not block startup since discovery takes 31s or more
+    asyncio.create_task(_async_discovery())
+
+    async_track_time_interval(hass, _async_discovery, DISCOVERY_INTERVAL)
+
+
+async def async_discover_devices(hass: HomeAssistant) -> list[UnifiDevice]:
```

Comment:
We don't use `hass`.

Reply:
https://github.com/home-assistant/core/pull/64413